File: Microsoft/10. Shortest Unsorted Continuous Subarray/code.py
class Solution(object):
    def findUnsortedSubarray(self, nums):
        l=nums[:]
        l.sort()
        count,left,right=0,0,0
        for i in range(len(nums)):
            if l[i]!=nums[i]:
                left = i
                break
        for i in range(len(nums)-1,-1,-1):
            if l[i]!=nums[i]:
                right = i
                break
        if left==0 and right==0:
            return 0
        elif left==right:
            return 1
        return right-left+1

File: Goldman Sachs/12. All Elements in Two Binary Search Trees/code.cpp
class Solution {
public:
    void inorder(TreeNode* root, vector<int> &ans) {
        if(root) {
            inorder(root->left,ans);
            ans.push_back(root->val);
            inorder(root->right,ans);
        }
        else return;
    }
    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        vector<int> ans;
        inorder(root1,ans);
        inorder(root2,ans);
        sort(ans.begin(),ans.end());
        return ans;

    }
};
